ui/app/helpers/utils/token-util.js

```diff
@@ -200,3 +201,30 @@ export function getFormattedTokenFiatAmount (
   })
   return `${formatCurrency(currentTokenInFiat, currentCurrency)} ${currentCurrency.toUpperCase()}`
 }
+
+/**
+ * Given a transaction receipt for a contract transaction that involves a token transfer to a given account,
+ * returns the amount of tokens transferred.
+ *
+ * @param {object} [txReceipt] - The transaction receipt object, as can be found in any standard MetaMask txMeta object
+ * @param {string} tokenAddres - The hex address of the token being transferred
+ * @param {string} receiverAddress - The address of the account being transferred to
+ * @param {number} [tokenDecimals] - The decimals of the token being transfered
+ * @returns {string|null} The amount of tokens transferred as a decimal number
+ */
+export function getTokensRecivedFromTxReceipt (txReceipt, tokenAddress, receiverAddress, tokenDecimals) {
+  const txReceiptLogs = txReceipt?.logs
+  if (txReceiptLogs && txReceipt?.status !== '0x0') {
```

Comment:
Due to the use of optional chaining, an undefined `txReceipt` would result in the second half of this condition evaluating to `true` (because `undefined !== '0x0'`). This isn't especially disastrous in this case because the first half of the condition would always fail in this case, but still this is probably not what was intended.

The `status` property should always be set to `0x1` for a successful transaction, right? So perhaps checking for the success case would be clearer:

```suggestion
  if (txReceiptLogs && txReceipt.status === '0x1') {
```

Or we could go the more verbose route:

```suggestion
  if (txReceipt && txReceiptLogs && txReceipt.status !== '0x0') {
```